### PR TITLE
DEVEXP-119: JS Debug Tests are Failing

### PR DESCRIPTION
**Pull request by BillFarber**
This is primarily an attempt to make things more clear. Especially how to run the tests.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -72,7 +72,62 @@ It is recommended to use VSCode as the editor for this package, as it can self-l
...
+    * Requires ML App Server setup first (See "Debugger Integration Testing" below for more information)
+    * Runs the tests under /dist/test/integration
+    * Transpiled from files under /client/test/integration
+        * sjsAdapter.test.js 
+
+#### Debugger Integration Testing
+
+Testing the JavaScript adapter requires a running MarkLogic server, so testing will not cover adapter by default.
+
+*** 3/1/23 - Unable to verify this -> *** To enable, set the flag in `runSJSDebuggerTest` to `true` in [**suite/index.ts**](suite/index.ts).
```
Comment (rjrudin) [inline CONTRIBUTING.md]: If this flag is gone, just remove this entire line.

**Review by rjrudin — APPROVED** [review]
This is good enough for me if the tests are all passing on your machine. Would be good to know if the tests pass for @bshalke too.

**Comment by bshalke** [comment]
I have one failing test in client:

```
  1 failing
  1) Extension Test Suite
       getDbClient should cache its settings in the state passed to it:

      AssertionError [ERR_ASSERTION]: Values have same structure but are not reference-equal:

MarklogicClient {
  mldbClient: <ref *1> MarkLogicClient {
    config: {
      extlibs: ExtLibs {
        client: [Circular *1]
      },
      patch: {
        replace: ExtlibsWrapper {
          dir: '/marklogic/patch/apply/',
          extlibs: ExtLibs {
            client: [Circular *1]
          },
          name: 'patch replace'
        }
      },
      query: {
        custom: ExtlibsWrapper {
          dir: '/marklogic/query/custom/',
          extlibs: ExtLibs {
            client: [Circular *1]
          },
          name: 'custom query'
        },
        snippet: ExtlibsWrapper {
          dir: '/marklogic/snippet/custom/',
          extlibs: ExtLibs {
            client: [Circular *1]
          },
          name: 'query snippet'
        }
      },
      resources: Resources {
        client: [Circular *1]
      },
      serverprops: RESTServerProperties {
        client: [Circular *1]
      },
      transforms: Transforms {
        client: [Circular *1]
      }
    },
    connectionParams: {
      agent: Agent {
        [Symbol(kCapture)]: false,
        _events: [Object: null prototype] {
          free: [Function (anonymous)],
...

      + expected - actual


      at Context.<anonymous> (c:\Users\bshal\OneDrive\Documents\GitHub\mlxprs\client\test\suite\client.test.ts:36:16)
      at processImmediate (node:internal/timers:466:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

Error: 1 tests failed.
    at c:\Users\bshal\OneDrive\Documents\GitHub\mlxprs\client\test\suite\index.ts:30:27
    at done (c:\Users\bshal\OneDrive\Documents\GitHub\mlxprs\node_modules\mocha\lib\mocha.js:1011:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

One test fails in client integration:

```
 1 failing
extensionHostProcess.js:100
  1) JavaScript Debug Test Suite
       Testing sjs/xqy boundary in eval/invoke
         sjs calling xdmp:invoke():
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (c:\Users\bshal\OneDrive\Documents\GitHub\mlxprs\dist\test\integration\sjsAdapter.test.js)
      at listOnTimeout (node:internal/timers:559:17)
      at processTimers (node:internal/timers:502:7)

extensionHostProcess.js:100
Error: 1 tests failed.
    at c:\Users\bshal\OneDrive\Documents\GitHub\mlxprs\client\test\integration\index.ts:31:27
    at done (c:\Users\bshal\OneDrive\Documents\GitHub\mlxprs\node_modules\mocha\lib\mocha.js:1011:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Server tests all pass

This is going off of latest develop branch.

**Comment by BillFarber** [comment]
@bshalke - I'm finding that the client integration tests are not 100% reliable. I think it's a timeout issue. My guess is that if you run it again, it will pass. I'm hoping to figure this part out soon.

For the client test that failed, I haven't seen that and I have no idea what's happening.

**Comment by BillFarber** [comment]
One other thought, Bryan.
With my next PR, I have made some changes and added some instructions to make it easier to run all 3 test sets from the command line. 
The instructions don't include this, but it might be worth starting from a clean slate, so I've been doing something like the following before testing:
`rm -rf  node_modules/ .vscode-test/ dist/ server/dist/ server/node_modules/ server/out/`

**Pull request merged**